### PR TITLE
Unify system call detection

### PR DESCRIPTION
**Pull request by hainest**
I'll refactor the interrupt detection in a separate PR.

### CODE REVIEW

File: instructionAPI/src/syscalls.C
Comment (kupsch) [inline instructionAPI/src/syscalls.C:1]: add dyninst copyright/license comment

**Review by kupsch — APPROVED** [review]
Looks good except missing comment in syscalls.C.

File: instructionAPI/src/syscalls.C
Comment (hainest) [reply instructionAPI/src/syscalls.C:1]: Why do I always forget that?

**Comment by hainest** [comment]
@kupsch I decided to go ahead and remove IA_IAPI::isSyscall here, as well.

**Review by kupsch — APPROVED** [review]

**Pull request merged**